homeassistant/components/hisense_connectlife/climate.py

```diff
@@ -0,0 +1,777 @@
+    SWING_HORIZONTAL,
+    SWING_OFF,
+    SWING_VERTICAL,
+)
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.const import ATTR_TEMPERATURE, UnitOfTemperature
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers.entity import DeviceInfo
+from homeassistant.helpers.entity_platform import AddEntitiesCallback
+from homeassistant.helpers.update_coordinator import CoordinatorEntity
+
+from .const import (
+    DOMAIN,
+    FAN_AUTO,
+    FAN_HIGH,
+    FAN_LOW,
+    FAN_MEDIUM,
+    FAN_ULTRA_HIGH,
+    FAN_ULTRA_LOW,
+    MAX_TEMP,
+    MIN_TEMP,
+    SFAN_ULTRA_HIGH,
+    SFAN_ULTRA_LOW,
+    StatusKey,
+)
+from .coordinator import HisenseACPluginDataUpdateCoordinator
+from .models import DeviceInfo as HisenseDeviceInfo
+
+_LOGGER = logging.getLogger(__name__)
+
+
+async def async_setup_entry(
+    hass: HomeAssistant,
+    config_entry: ConfigEntry,
+    async_add_entities: AddEntitiesCallback,
+) -> None:
+    """Set up the Hisense AC climate platform."""
+    coordinator: HisenseACPluginDataUpdateCoordinator = config_entry.runtime_data
+
+    try:
+        # Trigger initial data update
+        await coordinator.async_config_entry_first_refresh()
+
+        # Get devices from coordinator
+        devices = coordinator.data
+        _LOGGER.debug("Coordinator data after refresh: %s", devices)
+
+        if not devices:
+            _LOGGER.warning("No devices found in coordinator data")
+            return
+
+        entities = []
+        for device in devices.values():
+            _LOGGER.debug("Processing device: %s", device.to_dict())
+            if isinstance(device, HisenseDeviceInfo) and device.is_supported():
+                _LOGGER.info(
+                    "Adding climate entity for device: %s (type: %s-%s)",
+                    device.name,
+                    device.type_code,
+                    device.feature_code,
+                )
+                entity = HisenseClimate(coordinator, device)
+                entities.append(entity)
+            else:
+                _LOGGER.warning(
+                    "Skipping unsupported device: %s-%s (%s)",
+                    getattr(device, "type_code", None),
+                    getattr(device, "feature_code", None),
+                    getattr(device, "name", None),
+                )
+
+        if not entities:
+            _LOGGER.warning("No supported devices found")
+            return
+
+        _LOGGER.info("Adding %d climate entities", len(entities))
+        async_add_entities(entities)
+
+    except Exception as err:
+        _LOGGER.error("Failed to set up climate platform: %s", err)
+        raise
+
+
+class HisenseClimate(CoordinatorEntity, ClimateEntity):
+    """Hisense AC climate entity."""
+
+    _attr_has_entity_name = False
+    _attr_temperature_unit = UnitOfTemperature.CELSIUS
+    _attr_min_temp = MIN_TEMP
+    _attr_max_temp = MAX_TEMP
+    _attr_target_temperature_step = 1
+    _attr_supported_features = (
```

Comment:
Are these true for any and all climate devices - or could they be specific to a device (if so you should move them into your init instead.

Reply:
Reading further, the temperature step shouldn't be there as you adjust it already in init :)